pwndbg/dbg/gdb/__init__.py

```diff
@@ -430,54 +430,70 @@ def vmmap(self) -> pwndbg.dbg_mod.MemoryMap:
         pages.sort()
         return GDBMemoryMap(qemu, pages)
 
+    def _find_memory_last_readable(self, start: int, count: int) -> int:
+        end = start + count
+        result = -1
+
+        def _is_readable(addr) -> bool:
```

Comment:
Te function object is constructed on each execution of this function. Since it does not use any closure, pls put it outside of _find_memory_last_readable, or inline its logic below.